test/e2e/lib/flows/login-flow.js

```diff
@@ -95,6 +96,13 @@ export default class LoginFlow {
 		}
 
 		await loginPage.login( this.account.email || this.account.username, this.account.password );
+
+		if ( process.env.FLAGS === 'nav-unification' ) {
+			// Makes sure that the nav-unification welcome modal will be dismissed.
+			const guideComponent = new GuideComponent( this.driver, 1000 );
+			await guideComponent.dismissModal( '.nav-unification-modal' );
```

Comment:
We can probably replace `.nav-unification-modal` with `.components-modal__frame` which is the `<Guide />` components main class name as it's likely to be more reliable.

Reply:
This is answered here https://github.com/Automattic/wp-calypso/pull/50195#discussion_r578489363